### PR TITLE
improve test ecosystem

### LINKED ISSUE #147: Test compile fail
I was getting an error with `cargo test` on latest main:
```
error[E0432]: unresolved import `rinex::test_toolkit`
 --> rinex/tests/compression.rs:4:9
  |
4 |     use rinex::test_toolkit;
  |         ^^^^^^^^^^^^^^^^^^^ no `test_toolkit` in the root

For more information about this error, try `rustc --explain E0432`
```

I had to add `+#[cfg(feature = "tests")]` to `rinex/tests/compression.rs` and `rinex/tests/production.rs` to get it to run the tests.

Also the tests left two "untracked" git files `rinex/merge.rnx` `rinex/test.crx`. It would be nice to either add these to `.gitignore`, or two write test output files to a temporary directory so not to touch the source tree.

### PR DESCRIPTION
**Pull request by gwbres**
- [x] fix #147 
- [x] move `rinex/tests` within the crate (`rinex/src/tests` as a module)
- [x] ignore files generated by "doc rs"
- [x] use `std::PathBuf` more often

### CODE REVIEW

File: rinex/src/tests/compression.rs [hunk 1]
```diff
@@ -1,7 +1,8 @@
 #[cfg(test)]
 mod test {
-    use rinex::prelude::*;
-    use rinex::test_toolkit;
+    use crate::prelude::*;
+    use crate::tests::toolkit::*;
+    use std::path::PathBuf;
     //#[test]
     fn crinex1() {
```
Comment (lnicola) [inline rinex/src/tests/compression.rs:7]: Is this intentionally disabled?

Comment (gwbres) [reply rinex/src/tests/compression.rs:7]: Yes, there are still some issues in the RNX->CRNX operation.  
This test partially passes, and partially fails, and at the time I decided to temporarily disable it.
Will get back to file operations when I've accomplished what I want with file processing

Comment (lnicola) [reply rinex/src/tests/compression.rs:7]: You can use `#[ignore]`, it's a little more explicit.

File: rinex/src/tests/merge.rs [hunk 2]
```diff
@@ -0,0 +1,79 @@
+#[cfg(test)]
+mod merge {
```
[the remark this answers is not in the document]
Comment (gwbres) [reply rinex/src/tests/merge.rs:2]: @lnicola, 
don't know why this one was named `merge` as I usually use `test`

File: rinex/src/tests/meteo.rs [hunk 3]
```diff
@@ -1,7 +1,7 @@
 #[cfg(test)]
 #[cfg(feature = "meteo")]
```
[the remark this answers is not in the document]
Comment (gwbres) [reply rinex/src/tests/meteo.rs:2]: very true 👍

File: rinex/src/tests/compression.rs (hunk 1 above)
Comment (gwbres) [reply rinex/src/tests/compression.rs:7]: that's a cool asset, thanks

**Review by lnicola — APPROVED** [review]
Approving now so you can merge, I'll be less responsive today.

**Review by larsnaesbye — APPROVED** [review]

File: rinex/src/tests/compression.rs (hunk 1 above)
Comment (larsnaesbye) [reply rinex/src/tests/compression.rs:8]: Resolving this conversation to facilitate merging :)

**Comment by gwbres** [comment]
Thank you guys

**Pull request merged**